### PR TITLE
Feature/idp integration

### PR DESCRIPTION
**Pull request by olye0216**

### CODE REVIEW

File: xm-commons-idp/HELP.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## xm-commons-idp
+XM^online commons project for XM authentication context. 
+// FIXME: rename file to README.md if it will have some useful content or delete
```
Comment (amedvedchuk) [inline xm-commons-idp/HELP.md:3]: @olye0216 please resolve fixme - rename file to README.md

File: xm-commons-idp/src/main/java/com/icthh/xm/commons/repository/JwksRepository.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    }
+
+    private String extractKeyFromPath(String configPath, String keyName) {
+        return matcher
+            .extractUriTemplateVariables(PUBLIC_JWKS_CONFIG_PATTERN, configPath)
+            .get(keyName);
+    }
+
+    public Map<String, String> getTenantIdpJwks() {
```
Comment (amedvedchuk) [inline xm-commons-idp/src/main/java/com/icthh/xm/commons/repository/JwksRepository.java:83]: do we need this method? it is not used.

File: build.gradle [hunk 3]
```diff
@@ -70,6 +70,7 @@ subprojects {
     repositories {
         mavenCentral()
         jcenter()
+        maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }//TODO remove
```
Comment (vkirichenko) [inline build.gradle:73]: will be removed after commons merging

File: xm-commons-lep/src/main/java/com/icthh/xm/commons/lep/XmLepResourceService.java [hunk 4]
```diff
@@ -219,7 +220,12 @@ private String getTenantScriptLocation(String path, ContextsHolder contextsHolde
 
             case FILE: {
                 String lepDir = Paths.get(FileSystemUtils.APP_HOME_DIR, "config", "tenants",
-                                          tenantKey.toUpperCase(), appName, "lep").toString();
+                    tenantKey.toUpperCase(), appName, "lep").toString();
+
+                if (SystemUtils.IS_OS_WINDOWS) {
+                    return "file:///" + lepDir + path;
```
Comment (sergeysenja1992) [inline xm-commons-lep/src/main/java/com/icthh/xm/commons/lep/XmLepResourceService.java:226]: user methods from lib

File: xm-commons-idp/HELP.md (hunk 1 above)
Comment (olye0216) [reply xm-commons-idp/HELP.md:3]: Remark applied.

File: xm-commons-idp/src/main/java/com/icthh/xm/commons/repository/JwksRepository.java (hunk 2 above)
Comment (olye0216) [reply xm-commons-idp/src/main/java/com/icthh/xm/commons/repository/JwksRepository.java:83]: This method currently used in uaa MS.

File: xm-commons-lep/src/main/java/com/icthh/xm/commons/lep/XmLepResourceService.java (hunk 4 above)
Comment (olye0216) [reply xm-commons-lep/src/main/java/com/icthh/xm/commons/lep/XmLepResourceService.java:226]: Please concretize about what methods from lib you are talking about and where exactly they should be used.

File: xm-commons-idp/src/main/java/com/icthh/xm/commons/repository/JwksRepository.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    @Override
+    public void onInit(String configKey, String configValue) {
+        processJwks(configKey, configValue);
+    }
+
+    @Override
+    public void onRefresh(String updatedKey, String config) {
+        processJwks(updatedKey, config);
```
[the remark this answers is not in the document]
Comment (olye0216) [reply xm-commons-idp/src/main/java/com/icthh/xm/commons/repository/JwksRepository.java:63]: Remark applied.

File: xm-commons-lep/src/main/java/com/icthh/xm/commons/lep/XmLepResourceService.java (hunk 4 above)
Comment (sergeysenja1992) [reply xm-commons-lep/src/main/java/com/icthh/xm/commons/lep/XmLepResourceService.java]: any, for example
https://commons.apache.org/proper/commons-io/javadocs/api-1.4/org/apache/commons/io/FilenameUtils.html

Comment (olye0216) [reply xm-commons-lep/src/main/java/com/icthh/xm/commons/lep/XmLepResourceService.java]: Remark applied.

Comment (olye0216) [reply xm-commons-lep/src/main/java/com/icthh/xm/commons/lep/XmLepResourceService.java]: remark applied.

**Review by vkirichenko — APPROVED** [review]

**Review by amedvedchuk — APPROVED** [review]

**Pull request merged**